src/evaluate.jl

```diff
@@ -194,26 +194,33 @@ it's evaluated at zero. Note that the syntax `a(vals)` is equivalent to
 `evaluate(a, vals)`; and `a()` is equivalent to `evaluate(a)`.
 """
 function evaluate(a::HomogeneousPolynomial{T}, vals::NTuple{N,S} ) where
-        {T<:Number, S<:NumberNotSeriesN, N}
+        {T<:Number, S<:Number, N}
 
     @assert N == get_numvars()
 
+    return _evaluate(a, vals)
+end
+
+function _evaluate(a::HomogeneousPolynomial{T}, vals::NTuple{N,S} ) where
+        {T<:Number, S<:Number, N}
+
     ct = coeff_table[a.order+1]
     R = promote_type(T,S)
     suma = zero(R)
 
     for (i,a_coeff) in enumerate(a.coeffs)
         iszero(a_coeff) && continue
-        tmp = vals[1]^(ct[i][1])
+        @inbounds tmp = vals[1]^(ct[i][1])
         for n in 2:N
-            tmp *= vals[n]^(ct[i][n])
+            @inbounds tmp *= vals[n]^(ct[i][n])
```

Comment:
This loop could be written equivalently as `prod(vals .^ ct[i])`

Reply:
Thanks for the suggestion. I will change it as suggested.

Reply:
Done in [03e448e](/JuliaDiff/TaylorSeries.jl/pull/195/commits/03e448edd524aedfbfa8dc844621dbccd0d8260d).